cmake/ssl.cmake

```diff
@@ -35,6 +35,10 @@ elseif("${gRPC_SSL_PROVIDER}" STREQUAL "package")
   # We expect to locate OpenSSL using the built-in cmake module as the openssl
   # project itself does not provide installation support in its CMakeLists.txt
   # See https://cmake.org/cmake/help/v3.6/module/FindOpenSSL.html
+  if (APPLE AND (NOT OPENSSL_ROOT_DIR))
+    message(STATUS "Set OPENSSL_ROOT_DIR to /usr/local/opt/openssl on Apple platform")
```

Comment:
To be honest, this change seems pretty ad-hoc (at least without more context). Why /usr/local/opt/openssl and not something else?

Reply:
If find_package(OpenSSL ) doesn't work correctly on OS X (it looks like it might be a problem just on your machine, not in general), you can always use OPENSSL_ROOT_DIR to configure your build correctly.

Reply:
The preinstall openssl version on OS X is too old and not going to working for grpc, at least for latest version in master. And /usr/local/opt/openssl is the default path of `brew upgrade/install openssl`.
I admit it is not perfect. Maybe we can improve a little bit:

```
  find_package(OpenSSL REQUIRED)
  if (NOT OPENSSL_FOUND AND (APPLE AND (NOT OPENSSL_ROOT_DIR)))
    message(STATUS "Set OPENSSL_ROOT_DIR to /usr/local/opt/openssl on Apple platform")
    set (OPENSSL_ROOT_DIR "/usr/local/opt/openssl")
    find_package(OpenSSL REQUIRED)
  endif ()


```